lib/IRGen/IRGenSIL.cpp

```diff
@@ -1163,8 +1163,42 @@ class IRGenSILFunction :
           emitPackCountDebugVariable(Shape);
         }
       });
+
+    if (auto *BGT = llvm::dyn_cast<BoundGenericType>(Ty)) {
+      auto Decl = BGT->getDecl();
+      auto GE = Decl->getGenericEnvironment();
+      auto Requirements = BGT->getDecl()
+                              ->getGenericEnvironment()
+                              ->getGenericSignature()
+                              .getRequirements();
+      for (auto Requirement : Requirements) {
+        if (Requirement.getKind() == RequirementKind::Conformance) {
+          auto ProtocolDecl = Requirement.getProtocolDecl();
+          auto ConformingType = Requirement.getFirstType();
+          Type Archetype;
+          if (auto GTPT = llvm::dyn_cast<GenericTypeParamType>(
+                  ConformingType.getPointer()))
+            Archetype = GE->mapTypeIntoContext(GTPT);
+          else if (auto DMT = llvm::dyn_cast<DependentMemberType>(
+                       ConformingType.getPointer()))
+            Archetype = GE->mapTypeIntoContext(DMT);
```

Comment:
Howver, I'm not sure what this code is trying to do, but it's not correct. The types `Requirement` are written with the generic signature of the `BoundGenericType`. The `GE` you have here on the other hand is a generic environment for the generic signature of the current function. So the mapTypeIntoContext() call will just return an `ErrorType`.

Also, you're not looking at the _generic arguments_ of the BGT here. Presumably it should do that?